src/Boolean.php

```diff
@@ -4,31 +4,30 @@
 
 use InvalidArgumentException;
 
+use function Assert\that;
+
 class Boolean
 {
+    /**
+     * @var bool|null
+     */
     private $value;
 
     public function __construct($value, $default = null)
     {
-        static $empty_values = [null, ''];
-
-        if (is_bool($default) && in_array($value, $empty_values, true)) {
+        if ($value === null) {
             $value = $default;
         }
 
-        if (!is_scalar($value)) {
-            throw new InvalidArgumentException('Value must a possible boolean');
-        }
+        that($value)->nullOr()->scalar();
```

Comment:
personally I think it reads better as `Assertion::that($value)->nullOr()->scalar();`

"that" by itself seems to lack some context. If you want to keep the `use` maybe alias it?